hydrolib/tools/extforce_convert/converters.py

```diff
@@ -759,6 +798,113 @@ def contains(quantity_class, quantity) -> bool:
         return True
 
 
+class CmpToForcingConverter:
+    """
+    A class to convert CmpModel data into ForcingModel data for boundary condition definitions.
+    """
+
+    @staticmethod
+    def convert(cmp_model: CmpModel) -> ForcingModel:
+        """
+        Convert a CmpModel into a ForcingModel.
+
+        Args:
+            cmp_model (CmpModel):
+                The input CmpModel to be converted.
+
+        Returns:
+            ForcingModel: The converted ForcingModel.
+
+        Raises:
+            ValueError: If the lengths of the columns in the first row of the CmpModel do not match the number of
+                quantities in the CmpModel.
+
+        Examples:
+            Convert a CmpModel into a ForcingModel.
+                ```python
+                >>> from hydrolib.core.dflowfm.cmp.models import CmpModel
+                >>> from hydrolib.tools.ext_old_to_new.converters import CmpToForcingConverter
+                >>> data = {
+                ...     "comments": ["# Example comment"],
+                ...     "components": [
+                ...         {
+                ...             "harmonics": [{"period": 0.0, "amplitude": 1.0, "phase": 2.0}],
+                ...             "quantity_name": "harmonicboundary"
+                ...         }, {
+                ...             "astronomics": [{"name": "4MS10", "amplitude": 1.0, "phase": 2.0}],
+                ...             "quantity_name": "astronomicboundary"
+                ...         }
+                ...     ]
+                ... }
+                >>> cmp_model = CmpModel(**data)
+                >>> forcing_model = CmpToForcingConverter.convert(cmp_model)
+                >>> forcing_model.forcing[0].datablock
+                [[0.0, 1.0, 2.0]]
+                >>> forcing_model.forcing[1].datablock
+                [['4MS10', 1.0, 2.0]]
+
+                ```
+        """
+        forcing_list = []
+
+        for component in cmp_model.components:
+            harmonic_model = CmpToForcingConverter.convert_harmonic(
+                component.quantity_name, component.harmonics
+            )
+            if harmonic_model:
+                forcing_list.append(harmonic_model)
+
+            astronomic_model = CmpToForcingConverter.convert_astronomic(
+                component.quantity_name, component.astronomics
+            )
+            if astronomic_model:
+                forcing_list.append(astronomic_model)
+
+        return ForcingModel(forcing=forcing_list)
```

Comment:
can you return it as a forcing list (do not create the `ForcingModel` it will be created in the main boundaryCondition converter).